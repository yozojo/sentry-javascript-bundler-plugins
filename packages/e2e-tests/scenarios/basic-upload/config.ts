import { Options } from "@sentry/bundler-plugin-core";
import * as path from "path";

/**
 * The Sentry bundler plugin config object used for this test
 */
export const pluginConfig: Options = {
  release: {
    name: "basic-upload",
    uploadLegacySourcemaps: path.resolve(__dirname, "out"),
  },
  authToken: process.env["SENTRY_AUTH_TOKEN"] || "",
  org: "sentry-sdks",
  project: "js-bundler-plugin-e2e-tests",
};
