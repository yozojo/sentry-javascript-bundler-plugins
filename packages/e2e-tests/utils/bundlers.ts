export const BUNDLERS = ["rollup", "vite", "esbuild", "webpack4", "webpack5"];
