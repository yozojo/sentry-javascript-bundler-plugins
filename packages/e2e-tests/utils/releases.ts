import { AxiosError } from "axios";
import { BUNDLERS } from "./bundlers";
import {
  deleteReleaseFromSentry,
  getReleaseFileFromSentry,
  getReleaseFilesFromSentry,
} from "./sentry-api";

type ReleaseFilesData = {
  id: string;
  name: string;
  dist?: string;
  headers: Record<string, string>;
  size: number;
  sha1: string;
  dateCreated: string;
};

type ReleaseFile = {
  name: string;
  content: string;
};

export function deleteAllReleases(release: string) {
  return Promise.all(
    BUNDLERS.map(async (bundler) => {
      const bundlerRelease = `${release}-${bundler}`;
      try {
        const response = await deleteReleaseFromSentry(bundlerRelease);
        return response;
      } catch (e) {
        if ((e as AxiosError).response?.status === 404) {
          return Promise.resolve();
        }
        throw e;
      }
    })
  );
}

export async function getSentryReleaseFiles(release: string): Promise<ReleaseFile[]> {
  const releaseFileEntries = await getReleaseFiles(release);

  const releaseFiles: ReleaseFile[] = await Promise.all(
    releaseFileEntries.map((entry) => getReleaseFile(release, entry))
  );
  return releaseFiles;
}

async function getReleaseFiles(release: string): Promise<ReleaseFilesData[]> {
  const response = await getReleaseFilesFromSentry(release);
  return response.data as ReleaseFilesData[];
}

async function getReleaseFile(release: string, fileEntry: ReleaseFilesData): Promise<ReleaseFile> {
  const response = await getReleaseFileFromSentry(release, fileEntry.id);
  const data = response.data as ReleaseFile;

  return {
    name: fileEntry.name,
    // source maps are JSON and therefore, axios returns them as an object
    // We want them as a string, though, so we convert it
    content: typeof data === "object" ? JSON.stringify(data) : data,
  };
}
