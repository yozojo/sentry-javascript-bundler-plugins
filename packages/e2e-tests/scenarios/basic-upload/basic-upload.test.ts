import { pluginConfig } from "./config";
import { BUNDLERS } from "../../utils/bundlers";
import { getSentryReleaseFiles } from "../../utils/releases";

describe("Simple Sourcemaps Upload (one string include + default options)", () => {
  it.each(BUNDLERS)("uploads the correct files using %s", async (bundler) => {
    const release = `${pluginConfig.release?.name || ""}-${bundler}`;

    const sentryFiles = await getSentryReleaseFiles(release);

    expect(sentryFiles).toMatchSnapshot();
  });
});
