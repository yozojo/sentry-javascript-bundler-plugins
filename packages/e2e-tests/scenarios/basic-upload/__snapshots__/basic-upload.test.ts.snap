// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Sourcemaps Upload (one string include + default options) uploads the correct files using esbuild 1`] = `
Array [
  Object {
    "content": "\\"use strict\\";var i=typeof window<\\"u\\"?window:typeof global<\\"u\\"?global:typeof self<\\"u\\"?self:{};i.SENTRY_RELEASE={id:\\"basic-upload-esbuild\\"};var e=o=>{if(o===3)throw new Error(\\"I'm an uncaught error\\");return o<=1?o:e(o-1)+e(o-2)};console.log(\\"Hi, I'm a very simple app\\");e(10);console.log(\\"I'm done\\");
//# sourceMappingURL=index.js.map
",
    "name": "~/index.js",
  },
  Object {
    "content": "{\\"version\\":3,\\"sources\\":[\\"../../../../../bundler-plugin-core/sentry-release-injection-file.js\\",\\"../../input/fib.js\\",\\"../../input/index.js\\"],\\"sourcesContent\\":[\\"// This const is used for nothing except to make this file identifiable via its content.\\\\n// We search for \\\\\\"_sentry_release_injection_file\\\\\\" in the plugin to determine for sure that the file we look at is the release injection file.\\\\n\\\\n// _sentry_release_injection_file\\\\n\\\\n    var _global =\\\\n      typeof window !== 'undefined' ?\\\\n        window :\\\\n        typeof global !== 'undefined' ?\\\\n          global :\\\\n          typeof self !== 'undefined' ?\\\\n            self :\\\\n            {};\\\\n\\\\n    _global.SENTRY_RELEASE={id:\\\\\\"basic-upload-esbuild\\\\\\"};\\",\\"export const fibonacci = (n) => {\\\\n  if (n === 3) {\\\\n    throw new Error(\\\\\\"I'm an uncaught error\\\\\\");\\\\n  }\\\\n  if (n <= 1) {\\\\n    return n;\\\\n  }\\\\n  return fibonacci(n - 1) + fibonacci(n - 2);\\\\n};\\\\n;\\\\nimport \\\\\\"/home/runner/work/sentry-javascript-bundler-plugins/sentry-javascript-bundler-plugins/packages/bundler-plugin-core/sentry-release-injection-file.js\\\\\\";\\",\\"import { fibonacci } from \\\\\\"./fib\\\\\\";\\\\nconsole.log(\\\\\\"Hi, I'm a very simple app\\\\\\");\\\\n\\\\nfibonacci(10);\\\\n\\\\nconsole.log(\\\\\\"I'm done\\\\\\");\\\\n;\\\\nimport \\\\\\"/home/runner/work/sentry-javascript-bundler-plugins/sentry-javascript-bundler-plugins/packages/bundler-plugin-core/sentry-release-injection-file.js\\\\\\";\\"],\\"names\\":[],\\"mappings\\":\\"aAKI,GAAI,GACF,MAAO,QAAW,IAChB,OACA,MAAO,QAAW,IAChB,OACA,MAAO,MAAS,IACd,KACA,CAAC,EAET,EAAQ,eAAe,CAAC,GAAG,sBAAsB,ECd9C,GAAM,GAAY,AAAC,GAAM,CAC9B,GAAI,IAAM,EACR,KAAM,IAAI,OAAM,uBAAuB,EAEzC,MAAI,IAAK,EACA,EAEF,EAAU,EAAI,CAAC,EAAI,EAAU,EAAI,CAAC,CAC3C,ECPA,QAAQ,IAAI,2BAA2B,EAEvC,EAAU,EAAE,EAEZ,QAAQ,IAAI,UAAU\\"}",
    "name": "~/index.js.map",
  },
]
`;

exports[`Simple Sourcemaps Upload (one string include + default options) uploads the correct files using rollup 1`] = `
Array [
  Object {
    "content": "'use strict';

// This const is used for nothing except to make this file identifiable via its content.
// We search for \\"_sentry_release_injection_file\\" in the plugin to determine for sure that the file we look at is the release injection file.

// _sentry_release_injection_file

    var _global =
      typeof window !== 'undefined' ?
        window :
        typeof global !== 'undefined' ?
          global :
          typeof self !== 'undefined' ?
            self :
            {};

    _global.SENTRY_RELEASE={id:\\"basic-upload-rollup\\"};

const fibonacci = (n) => {
  if (n === 3) {
    throw new Error(\\"I'm an uncaught error\\");
  }
  if (n <= 1) {
    return n;
  }
  return fibonacci(n - 1) + fibonacci(n - 2);
};

console.log(\\"Hi, I'm a very simple app\\");

fibonacci(10);

console.log(\\"I'm done\\");
//# sourceMappingURL=index.js.map
",
    "name": "~/index.js",
  },
  Object {
    "content": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sources\\":[\\"../../../../../bundler-plugin-core/sentry-release-injection-file.js\\",\\"../../input/fib.js\\",\\"../../input/index.js\\"],\\"sourcesContent\\":[\\"// This const is used for nothing except to make this file identifiable via its content.\\\\n// We search for \\\\\\"_sentry_release_injection_file\\\\\\" in the plugin to determine for sure that the file we look at is the release injection file.\\\\n\\\\n// _sentry_release_injection_file\\\\n\\",\\"export const fibonacci = (n) => {\\\\n  if (n === 3) {\\\\n    throw new Error(\\\\\\"I'm an uncaught error\\\\\\");\\\\n  }\\\\n  if (n <= 1) {\\\\n    return n;\\\\n  }\\\\n  return fibonacci(n - 1) + fibonacci(n - 2);\\\\n};\\\\n\\",\\"import { fibonacci } from \\\\\\"./fib\\\\\\";\\\\nconsole.log(\\\\\\"Hi, I'm a very simple app\\\\\\");\\\\n\\\\nfibonacci(10);\\\\n\\\\nconsole.log(\\\\\\"I'm done\\\\\\");\\\\n\\"],\\"names\\":[],\\"mappings\\":\\";;AAAA;AACA;AACA;AACA;;;;;;;;;;;;;ACHA,MAAA,SAAA,GAAA,CAAA,CAAA,KAAA;AACA,EAAA,IAAA,CAAA,KAAA,CAAA,EAAA;AACA,IAAA,MAAA,IAAA,KAAA,CAAA,uBAAA,CAAA,CAAA;AACA,GAAA;AACA,EAAA,IAAA,CAAA,IAAA,CAAA,EAAA;AACA,IAAA,OAAA,CAAA,CAAA;AACA,GAAA;AACA,EAAA,OAAA,SAAA,CAAA,CAAA,GAAA,CAAA,CAAA,GAAA,SAAA,CAAA,CAAA,GAAA,CAAA,CAAA,CAAA;AACA,CAAA;;ACPA,OAAA,CAAA,GAAA,CAAA,2BAAA,CAAA,CAAA;AACA;AACA,SAAA,CAAA,EAAA,CAAA,CAAA;AACA;AACA,OAAA,CAAA,GAAA,CAAA,UAAA,CAAA\\"}",
    "name": "~/index.js.map",
  },
]
`;

exports[`Simple Sourcemaps Upload (one string include + default options) uploads the correct files using vite 1`] = `
Array [
  Object {
    "content": "\\"use strict\\";var i=typeof window<\\"u\\"?window:typeof global<\\"u\\"?global:typeof self<\\"u\\"?self:{};i.SENTRY_RELEASE={id:\\"basic-upload-vite\\"};const o=e=>{if(e===3)throw new Error(\\"I'm an uncaught error\\");return e<=1?e:o(e-1)+o(e-2)};console.log(\\"Hi, I'm a very simple app\\");o(10);console.log(\\"I'm done\\");
//# sourceMappingURL=index.js.map
",
    "name": "~/index.js",
  },
  Object {
    "content": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sources\\":[\\"../../input/fib.js\\",\\"../../input/index.js\\"],\\"sourcesContent\\":[\\"export const fibonacci = (n) => {\\\\n  if (n === 3) {\\\\n    throw new Error(\\\\\\"I'm an uncaught error\\\\\\");\\\\n  }\\\\n  if (n <= 1) {\\\\n    return n;\\\\n  }\\\\n  return fibonacci(n - 1) + fibonacci(n - 2);\\\\n};\\\\n\\",\\"import { fibonacci } from \\\\\\"./fib\\\\\\";\\\\nconsole.log(\\\\\\"Hi, I'm a very simple app\\\\\\");\\\\n\\\\nfibonacci(10);\\\\n\\\\nconsole.log(\\\\\\"I'm done\\\\\\");\\\\n\\"],\\"names\\":[\\"fibonacci\\",\\"n\\"],\\"mappings\\":\\"uIAAA,MAAAA,EAAAC,GAAA,CACA,GAAAA,IAAA,EACA,MAAA,IAAA,MAAA,uBAAA,EAEA,OAAAA,GAAA,EACAA,EAEAD,EAAAC,EAAA,CAAA,EAAAD,EAAAC,EAAA,CAAA,CACA,ECPA,QAAA,IAAA,2BAAA,EAEAD,EAAA,EAAA,EAEA,QAAA,IAAA,UAAA\\"}",
    "name": "~/index.js.map",
  },
]
`;

exports[`Simple Sourcemaps Upload (one string include + default options) uploads the correct files using webpack4 1`] = `
Array [
  Object {
    "content": "!function(e,n){for(var r in n)e[r]=n[r]}(exports,function(e){var n={};function r(t){if(n[t])return n[t].exports;var o=n[t]={i:t,l:!1,exports:{}};return e[t].call(o.exports,o,o.exports,r),o.l=!0,o.exports}return r.m=e,r.c=n,r.d=function(e,n,t){r.o(e,n)||Object.defineProperty(e,n,{enumerable:!0,get:t})},r.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},r.t=function(e,n){if(1&n&&(e=r(e)),8&n)return e;if(4&n&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var t=Object.create(null);if(r.r(t),Object.defineProperty(t,\\"default\\",{enumerable:!0,value:e}),2&n&&\\"string\\"!=typeof e)for(var o in e)r.d(t,o,function(n){return e[n]}.bind(null,o));return t},r.n=function(e){var n=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(n,\\"a\\",n),n},r.o=function(e,n){return Object.prototype.hasOwnProperty.call(e,n)},r.p=\\"\\",r(r.s=1)}([function(e,n){(\\"undefined\\"!=typeof window?window:\\"undefined\\"!=typeof global?global:\\"undefined\\"!=typeof self?self:{}).SENTRY_RELEASE={id:\\"basic-upload-webpack4\\"}},function(e,n,r){\\"use strict\\";r.r(n);r(0);const t=e=>{if(3===e)throw new Error(\\"I'm an uncaught error\\");return e<=1?e:t(e-1)+t(e-2)};console.log(\\"Hi, I'm a very simple app\\"),t(10),console.log(\\"I'm done\\")}]));
//# sourceMappingURL=index.js.map",
    "name": "~/index.js",
  },
  Object {
    "content": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sources\\":[\\"webpack:///webpack/bootstrap\\",\\"webpack:///../bundler-plugin-core/sentry-release-injection-file.js\\",\\"webpack:///./scenarios/basic-upload/input/fib.js\\",\\"webpack:///./scenarios/basic-upload/input/index.js\\"],\\"sourcesContent\\":[\\" \\\\t// The module cache\\\\n \\\\tvar installedModules = {};\\\\n\\\\n \\\\t// The require function\\\\n \\\\tfunction __webpack_require__(moduleId) {\\\\n\\\\n \\\\t\\\\t// Check if module is in cache\\\\n \\\\t\\\\tif(installedModules[moduleId]) {\\\\n \\\\t\\\\t\\\\treturn installedModules[moduleId].exports;\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\t// Create a new module (and put it into the cache)\\\\n \\\\t\\\\tvar module = installedModules[moduleId] = {\\\\n \\\\t\\\\t\\\\ti: moduleId,\\\\n \\\\t\\\\t\\\\tl: false,\\\\n \\\\t\\\\t\\\\texports: {}\\\\n \\\\t\\\\t};\\\\n\\\\n \\\\t\\\\t// Execute the module function\\\\n \\\\t\\\\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\\\\n\\\\n \\\\t\\\\t// Flag the module as loaded\\\\n \\\\t\\\\tmodule.l = true;\\\\n\\\\n \\\\t\\\\t// Return the exports of the module\\\\n \\\\t\\\\treturn module.exports;\\\\n \\\\t}\\\\n\\\\n\\\\n \\\\t// expose the modules object (__webpack_modules__)\\\\n \\\\t__webpack_require__.m = modules;\\\\n\\\\n \\\\t// expose the module cache\\\\n \\\\t__webpack_require__.c = installedModules;\\\\n\\\\n \\\\t// define getter function for harmony exports\\\\n \\\\t__webpack_require__.d = function(exports, name, getter) {\\\\n \\\\t\\\\tif(!__webpack_require__.o(exports, name)) {\\\\n \\\\t\\\\t\\\\tObject.defineProperty(exports, name, { enumerable: true, get: getter });\\\\n \\\\t\\\\t}\\\\n \\\\t};\\\\n\\\\n \\\\t// define __esModule on exports\\\\n \\\\t__webpack_require__.r = function(exports) {\\\\n \\\\t\\\\tif(typeof Symbol !== 'undefined' && Symbol.toStringTag) {\\\\n \\\\t\\\\t\\\\tObject.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\tObject.defineProperty(exports, '__esModule', { value: true });\\\\n \\\\t};\\\\n\\\\n \\\\t// create a fake namespace object\\\\n \\\\t// mode & 1: value is a module id, require it\\\\n \\\\t// mode & 2: merge all properties of value into the ns\\\\n \\\\t// mode & 4: return value when already ns object\\\\n \\\\t// mode & 8|1: behave like require\\\\n \\\\t__webpack_require__.t = function(value, mode) {\\\\n \\\\t\\\\tif(mode & 1) value = __webpack_require__(value);\\\\n \\\\t\\\\tif(mode & 8) return value;\\\\n \\\\t\\\\tif((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;\\\\n \\\\t\\\\tvar ns = Object.create(null);\\\\n \\\\t\\\\t__webpack_require__.r(ns);\\\\n \\\\t\\\\tObject.defineProperty(ns, 'default', { enumerable: true, value: value });\\\\n \\\\t\\\\tif(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));\\\\n \\\\t\\\\treturn ns;\\\\n \\\\t};\\\\n\\\\n \\\\t// getDefaultExport function for compatibility with non-harmony modules\\\\n \\\\t__webpack_require__.n = function(module) {\\\\n \\\\t\\\\tvar getter = module && module.__esModule ?\\\\n \\\\t\\\\t\\\\tfunction getDefault() { return module['default']; } :\\\\n \\\\t\\\\t\\\\tfunction getModuleExports() { return module; };\\\\n \\\\t\\\\t__webpack_require__.d(getter, 'a', getter);\\\\n \\\\t\\\\treturn getter;\\\\n \\\\t};\\\\n\\\\n \\\\t// Object.prototype.hasOwnProperty.call\\\\n \\\\t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\\\\n\\\\n \\\\t// __webpack_public_path__\\\\n \\\\t__webpack_require__.p = \\\\\\"\\\\\\";\\\\n\\\\n\\\\n \\\\t// Load entry module and return exports\\\\n \\\\treturn __webpack_require__(__webpack_require__.s = 1);\\\\n\\",\\"// This const is used for nothing except to make this file identifiable via its content.\\\\n// We search for \\\\\\"_sentry_release_injection_file\\\\\\" in the plugin to determine for sure that the file we look at is the release injection file.\\\\n\\\\n// _sentry_release_injection_file\\\\n\\\\n    var _global =\\\\n      typeof window !== 'undefined' ?\\\\n        window :\\\\n        typeof global !== 'undefined' ?\\\\n          global :\\\\n          typeof self !== 'undefined' ?\\\\n            self :\\\\n            {};\\\\n\\\\n    _global.SENTRY_RELEASE={id:\\\\\\"basic-upload-webpack4\\\\\\"};\\",\\"export const fibonacci = (n) => {\\\\n  if (n === 3) {\\\\n    throw new Error(\\\\\\"I'm an uncaught error\\\\\\");\\\\n  }\\\\n  if (n <= 1) {\\\\n    return n;\\\\n  }\\\\n  return fibonacci(n - 1) + fibonacci(n - 2);\\\\n};\\\\n;\\\\nimport \\\\\\"/home/runner/work/sentry-javascript-bundler-plugins/sentry-javascript-bundler-plugins/packages/bundler-plugin-core/sentry-release-injection-file.js\\\\\\";\\",\\"import { fibonacci } from \\\\\\"./fib\\\\\\";\\\\nconsole.log(\\\\\\"Hi, I'm a very simple app\\\\\\");\\\\n\\\\nfibonacci(10);\\\\n\\\\nconsole.log(\\\\\\"I'm done\\\\\\");\\\\n;\\\\nimport \\\\\\"/home/runner/work/sentry-javascript-bundler-plugins/sentry-javascript-bundler-plugins/packages/bundler-plugin-core/sentry-release-injection-file.js\\\\\\";\\"],\\"names\\":[\\"installedModules\\",\\"__webpack_require__\\",\\"moduleId\\",\\"exports\\",\\"module\\",\\"i\\",\\"l\\",\\"modules\\",\\"call\\",\\"m\\",\\"c\\",\\"d\\",\\"name\\",\\"getter\\",\\"o\\",\\"Object\\",\\"defineProperty\\",\\"enumerable\\",\\"get\\",\\"r\\",\\"Symbol\\",\\"toStringTag\\",\\"value\\",\\"t\\",\\"mode\\",\\"__esModule\\",\\"ns\\",\\"create\\",\\"key\\",\\"bind\\",\\"n\\",\\"object\\",\\"property\\",\\"prototype\\",\\"hasOwnProperty\\",\\"p\\",\\"s\\",\\"window\\",\\"global\\",\\"self\\",\\"SENTRY_RELEASE\\",\\"id\\",\\"fibonacci\\",\\"Error\\",\\"console\\",\\"log\\"],\\"mappings\\":\\"6DACE,IAAIA,EAAmB,GAGvB,SAASC,EAAoBC,GAG5B,GAAGF,EAAiBE,GACnB,OAAOF,EAAiBE,GAAUC,QAGnC,IAAIC,EAASJ,EAAiBE,GAAY,CACzCG,EAAGH,EACHI,GAAG,EACHH,QAAS,IAUV,OANAI,EAAQL,GAAUM,KAAKJ,EAAOD,QAASC,EAAQA,EAAOD,QAASF,GAG/DG,EAAOE,GAAI,EAGJF,EAAOD,QA0Df,OArDAF,EAAoBQ,EAAIF,EAGxBN,EAAoBS,EAAIV,EAGxBC,EAAoBU,EAAI,SAASR,EAASS,EAAMC,GAC3CZ,EAAoBa,EAAEX,EAASS,IAClCG,OAAOC,eAAeb,EAASS,EAAM,CAAEK,YAAY,EAAMC,IAAKL,KAKhEZ,EAAoBkB,EAAI,SAAShB,GACX,oBAAXiB,QAA0BA,OAAOC,aAC1CN,OAAOC,eAAeb,EAASiB,OAAOC,YAAa,CAAEC,MAAO,WAE7DP,OAAOC,eAAeb,EAAS,aAAc,CAAEmB,OAAO,KAQvDrB,EAAoBsB,EAAI,SAASD,EAAOE,GAEvC,GADU,EAAPA,IAAUF,EAAQrB,EAAoBqB,IAC/B,EAAPE,EAAU,OAAOF,EACpB,GAAW,EAAPE,GAA8B,iBAAVF,GAAsBA,GAASA,EAAMG,WAAY,OAAOH,EAChF,IAAII,EAAKX,OAAOY,OAAO,MAGvB,GAFA1B,EAAoBkB,EAAEO,GACtBX,OAAOC,eAAeU,EAAI,UAAW,CAAET,YAAY,EAAMK,MAAOA,IACtD,EAAPE,GAA4B,iBAATF,EAAmB,IAAI,IAAIM,KAAON,EAAOrB,EAAoBU,EAAEe,EAAIE,EAAK,SAASA,GAAO,OAAON,EAAMM,IAAQC,KAAK,KAAMD,IAC9I,OAAOF,GAIRzB,EAAoB6B,EAAI,SAAS1B,GAChC,IAAIS,EAAST,GAAUA,EAAOqB,WAC7B,WAAwB,OAAOrB,EAAgB,SAC/C,WAA8B,OAAOA,GAEtC,OADAH,EAAoBU,EAAEE,EAAQ,IAAKA,GAC5BA,GAIRZ,EAAoBa,EAAI,SAASiB,EAAQC,GAAY,OAAOjB,OAAOkB,UAAUC,eAAe1B,KAAKuB,EAAQC,IAGzG/B,EAAoBkC,EAAI,GAIjBlC,EAAoBA,EAAoBmC,EAAI,G,iBC5E7B,oBAAXC,OACLA,OACkB,oBAAXC,OACLA,OACgB,oBAATC,KACLA,KACA,IAEAC,eAAe,CAACC,GAAG,0B,yCCdxB,MAAMC,EAAaZ,IACxB,GAAU,IAANA,EACF,MAAM,IAAIa,MAAM,yBAElB,OAAIb,GAAK,EACAA,EAEFY,EAAUZ,EAAI,GAAKY,EAAUZ,EAAI,ICN1Cc,QAAQC,IAAI,6BAEZH,EAAU,IAEVE,QAAQC,IAAI\\"}",
    "name": "~/index.js.map",
  },
]
`;

exports[`Simple Sourcemaps Upload (one string include + default options) uploads the correct files using webpack5 1`] = `
Array [
  Object {
    "content": "(()=>{var e={677:(e,o,r)=>{(\\"undefined\\"!=typeof window?window:void 0!==r.g?r.g:\\"undefined\\"!=typeof self?self:{}).SENTRY_RELEASE={id:\\"basic-upload-webpack5\\"}}},o={};function r(t){var n=o[t];if(void 0!==n)return n.exports;var i=o[t]={exports:{}};return e[t](i,i.exports,r),i.exports}r.g=function(){if(\\"object\\"==typeof globalThis)return globalThis;try{return this||new Function(\\"return this\\")()}catch(e){if(\\"object\\"==typeof window)return window}}(),r.r=e=>{\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})};var t={};(()=>{\\"use strict\\";r.r(t),r(677);const e=o=>{if(3===o)throw new Error(\\"I'm an uncaught error\\");return o<=1?o:e(o-1)+e(o-2)};console.log(\\"Hi, I'm a very simple app\\"),e(10),console.log(\\"I'm done\\")})();var n=exports;for(var i in t)n[i]=t[i];t.__esModule&&Object.defineProperty(n,\\"__esModule\\",{value:!0})})();
//# sourceMappingURL=index.js.map",
    "name": "~/index.js",
  },
  Object {
    "content": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sources\\":[\\"webpack://@sentry-internal/bundler-plugin-e2e-tests/../bundler-plugin-core/sentry-release-injection-file.js\\",\\"webpack://@sentry-internal/bundler-plugin-e2e-tests/webpack/bootstrap\\",\\"webpack://@sentry-internal/bundler-plugin-e2e-tests/webpack/runtime/global\\",\\"webpack://@sentry-internal/bundler-plugin-e2e-tests/webpack/runtime/make namespace object\\",\\"webpack://@sentry-internal/bundler-plugin-e2e-tests/./scenarios/basic-upload/input/fib.js\\",\\"webpack://@sentry-internal/bundler-plugin-e2e-tests/./scenarios/basic-upload/input/index.js\\"],\\"sourcesContent\\":[\\"// This const is used for nothing except to make this file identifiable via its content.\\\\n// We search for \\\\\\"_sentry_release_injection_file\\\\\\" in the plugin to determine for sure that the file we look at is the release injection file.\\\\n\\\\n// _sentry_release_injection_file\\\\n\\\\n    var _global =\\\\n      typeof window !== 'undefined' ?\\\\n        window :\\\\n        typeof global !== 'undefined' ?\\\\n          global :\\\\n          typeof self !== 'undefined' ?\\\\n            self :\\\\n            {};\\\\n\\\\n    _global.SENTRY_RELEASE={id:\\\\\\"basic-upload-webpack5\\\\\\"};\\",\\"// The module cache\\\\nvar __webpack_module_cache__ = {};\\\\n\\\\n// The require function\\\\nfunction __webpack_require__(moduleId) {\\\\n\\\\t// Check if module is in cache\\\\n\\\\tvar cachedModule = __webpack_module_cache__[moduleId];\\\\n\\\\tif (cachedModule !== undefined) {\\\\n\\\\t\\\\treturn cachedModule.exports;\\\\n\\\\t}\\\\n\\\\t// Create a new module (and put it into the cache)\\\\n\\\\tvar module = __webpack_module_cache__[moduleId] = {\\\\n\\\\t\\\\t// no module.id needed\\\\n\\\\t\\\\t// no module.loaded needed\\\\n\\\\t\\\\texports: {}\\\\n\\\\t};\\\\n\\\\n\\\\t// Execute the module function\\\\n\\\\t__webpack_modules__[moduleId](module, module.exports, __webpack_require__);\\\\n\\\\n\\\\t// Return the exports of the module\\\\n\\\\treturn module.exports;\\\\n}\\\\n\\\\n\\",\\"__webpack_require__.g = (function() {\\\\n\\\\tif (typeof globalThis === 'object') return globalThis;\\\\n\\\\ttry {\\\\n\\\\t\\\\treturn this || new Function('return this')();\\\\n\\\\t} catch (e) {\\\\n\\\\t\\\\tif (typeof window === 'object') return window;\\\\n\\\\t}\\\\n})();\\",\\"// define __esModule on exports\\\\n__webpack_require__.r = (exports) => {\\\\n\\\\tif(typeof Symbol !== 'undefined' && Symbol.toStringTag) {\\\\n\\\\t\\\\tObject.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });\\\\n\\\\t}\\\\n\\\\tObject.defineProperty(exports, '__esModule', { value: true });\\\\n};\\",\\"export const fibonacci = (n) => {\\\\n  if (n === 3) {\\\\n    throw new Error(\\\\\\"I'm an uncaught error\\\\\\");\\\\n  }\\\\n  if (n <= 1) {\\\\n    return n;\\\\n  }\\\\n  return fibonacci(n - 1) + fibonacci(n - 2);\\\\n};\\\\n;\\\\nimport \\\\\\"/home/runner/work/sentry-javascript-bundler-plugins/sentry-javascript-bundler-plugins/packages/bundler-plugin-core/sentry-release-injection-file.js\\\\\\";\\",\\"import { fibonacci } from \\\\\\"./fib\\\\\\";\\\\nconsole.log(\\\\\\"Hi, I'm a very simple app\\\\\\");\\\\n\\\\nfibonacci(10);\\\\n\\\\nconsole.log(\\\\\\"I'm done\\\\\\");\\\\n;\\\\nimport \\\\\\"/home/runner/work/sentry-javascript-bundler-plugins/sentry-javascript-bundler-plugins/packages/bundler-plugin-core/sentry-release-injection-file.js\\\\\\";\\"],\\"names\\":[\\"window\\",\\"g\\",\\"self\\",\\"SENTRY_RELEASE\\",\\"id\\",\\"__webpack_module_cache__\\",\\"__webpack_require__\\",\\"moduleId\\",\\"cachedModule\\",\\"undefined\\",\\"exports\\",\\"module\\",\\"__webpack_modules__\\",\\"globalThis\\",\\"this\\",\\"Function\\",\\"e\\",\\"r\\",\\"Symbol\\",\\"toStringTag\\",\\"Object\\",\\"defineProperty\\",\\"value\\",\\"fibonacci\\",\\"n\\",\\"Error\\",\\"console\\",\\"log\\"],\\"mappings\\":\\"4BAMwB,oBAAXA,OACLA,YACkB,IAAX,EAAAC,EACL,EAAAA,EACgB,oBAATC,KACLA,KACA,CAAC,GAEDC,eAAe,CAACC,GAAG,wB,GCb3BC,EAA2B,CAAC,EAGhC,SAASC,EAAoBC,GAE5B,IAAIC,EAAeH,EAAyBE,GAC5C,QAAqBE,IAAjBD,EACH,OAAOA,EAAaE,QAGrB,IAAIC,EAASN,EAAyBE,GAAY,CAGjDG,QAAS,CAAC,GAOX,OAHAE,EAAoBL,GAAUI,EAAQA,EAAOD,QAASJ,GAG/CK,EAAOD,OACf,CCtBAJ,EAAoBL,EAAI,WACvB,GAA0B,iBAAfY,WAAyB,OAAOA,WAC3C,IACC,OAAOC,MAAQ,IAAIC,SAAS,cAAb,EAGhB,CAFE,MAAOC,GACR,GAAsB,iBAAXhB,OAAqB,OAAOA,MACxC,CACA,CAPuB,GCCxBM,EAAoBW,EAAKP,IACH,oBAAXQ,QAA0BA,OAAOC,aAC1CC,OAAOC,eAAeX,EAASQ,OAAOC,YAAa,CAAEG,MAAO,WAE7DF,OAAOC,eAAeX,EAAS,aAAc,CAAEY,OAAO,GAAO,E,0CCLvD,MAAMC,EAAaC,IACxB,GAAU,IAANA,EACF,MAAM,IAAIC,MAAM,yBAElB,OAAID,GAAK,EACAA,EAEFD,EAAUC,EAAI,GAAKD,EAAUC,EAAI,EAAE,ECN5CE,QAAQC,IAAI,6BAEZJ,EAAU,IAEVG,QAAQC,IAAI,W\\"}",
    "name": "~/index.js.map",
  },
]
`;
